lib/ohai/plugins/hostname.rb

```diff
@@ -138,9 +138,23 @@ def sigar_is_available?
   end
 
   collect_data(:solaris2) do
+    require 'socket'
+
     machinename from_cmd("hostname")
     hostname from_cmd("hostname")
-    fqdn resolve_fqdn
+    fqdn_lookup = Socket.getaddrinfo(hostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first[2]
```

Comment:
can we just re-use resolve_fqdn here?  it looks like that has a rescue clause that you probably don't want to drop.